sopel/irc/__init__.py

```diff
@@ -47,11 +48,11 @@ class AbstractBot(abc.ABC):
     """Abstract definition of Sopel's interface."""
     def __init__(self, settings):
         # private properties: access as read-only properties
-        self._nick = tools.Identifier(settings.core.nick)
         self._user = settings.core.user
         self._name = settings.core.name
         self._isupport = ISupport()
         self._myinfo = None
+        self._nick = self.make_identifier(settings.core.nick)
```

Comment:
I don't like moving this to the bottom. The bot's nick is arguably its most important attribute, and it should stay grouped with the other identity-related stuff (user and name).

Reply:
It needs at least `self._isupport` first, so I figured I could put it there, given that it might, in the end, require more than that.

Reply:
Maybe "needs" is a bit strong at this point in the process? Sure, `self.isupport` will _exist_ but it won't be populated yet. Is there a meaningful, functional difference between this and the old way (with updated import)?

Reply:
In theory yes: if we change the default value for ISupport's CASEMAPPING, we may end up with a different result, which wasn't the case before. In fact, I should even move that *after* `settings` is set, so `make_identifiers()`  could take advantage of the configuration file.

However, I'm not sure to see the problem here: protected and undocumented attributes, all defined in the same bloc of code, very early in `__init__()`. It's not like there are 100 lines before the `self._nick` here.